parquet/src/test/java/org/apache/iceberg/parquet/TestDictionaryRowGroupFilter.java

```diff
@@ -79,16 +53,39 @@
 import org.apache.parquet.hadoop.metadata.ColumnChunkMetaData;
 import org.apache.parquet.hadoop.metadata.ColumnPath;
 import org.apache.parquet.schema.MessageType;
-import org.junit.Assume;
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.apache.iceberg.avro.AvroSchemaUtil.convert;
+import static org.apache.iceberg.expressions.Expressions.and;
+import static org.apache.iceberg.expressions.Expressions.equal;
+import static org.apache.iceberg.expressions.Expressions.greaterThan;
+import static org.apache.iceberg.expressions.Expressions.greaterThanOrEqual;
+import static org.apache.iceberg.expressions.Expressions.in;
+import static org.apache.iceberg.expressions.Expressions.isNaN;
+import static org.apache.iceberg.expressions.Expressions.isNull;
+import static org.apache.iceberg.expressions.Expressions.lessThan;
+import static org.apache.iceberg.expressions.Expressions.lessThanOrEqual;
+import static org.apache.iceberg.expressions.Expressions.not;
+import static org.apache.iceberg.expressions.Expressions.notEqual;
+import static org.apache.iceberg.expressions.Expressions.notIn;
+import static org.apache.iceberg.expressions.Expressions.notNaN;
+import static org.apache.iceberg.expressions.Expressions.notNull;
+import static org.apache.iceberg.expressions.Expressions.notStartsWith;
+import static org.apache.iceberg.expressions.Expressions.or;
+import static org.apache.iceberg.expressions.Expressions.startsWith;
+import static org.apache.iceberg.expressions.Expressions.truncate;
+import static org.apache.iceberg.types.Types.NestedField.optional;
+import static org.apache.iceberg.types.Types.NestedField.required;
+import static org.assertj.core.api.Assertions.assertThat;
 
-@RunWith(Parameterized.class)
-public class TestDictionaryRowGroupFilter {
+/**
+ * At time of development there's no out of the box way to implement a Parameterized BeforeEach in Junit5. 
```

Comment:
I don't think we actually want to create multiple classes for each parameter. I think it would make sense to check how easy/complicated it is to support class-level parameterization with JUnit5 as I know that's not supported by default

Reply:
maybe worth exploring whether https://github.com/junit-team/junit5/issues/3157#issuecomment-1435009129 would fit here

Reply:
Hi @lisirrx @nastra , I met the same problem when I tried to solve #9079. Can we refer to Flink here, introduce a [ParameterizedTestExtension](https://github.com/apache/flink/blob/master/flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/testutils/junit/extensions/parameterized/ParameterizedTestExtension.java)?

If you agree with me, I'd like to create a PR to do this.

Reply:
@wanglijie95 @nastra, ParameterizedTestExtension introduced in 4d381f840f926f7f4be3f871c6a62f8ca3f84406. 
I put it in an util test package, but I think the best idea should be to move these utils in a shared module.  Is it better to open a new issue to address it?

Reply:
+1, I think a shared module is better.
BTW, I will check if this approach fits the `iceberg-core` module.

Reply:
hi, @GianlucaPrincipini . Thanks for the `ParameterizedTestExtension ` which can used in most scenarios, but I encountered such case:
For example, In `org.apache.iceberg.TestDeleteFiles` class(a lot of other test classes do the same thing) which extends `org.apache.iceberg.TableTestBase`, the parameterized tests's parameters are passed through the constructor. A `formatVersion` as the constructor's param of  `TestDeleteFiles` will used in `TableTestBase`'s , and init some other field.

Maybe some changes in `ParameterizedTestExtension` are needed, and also does the '@Parameter' annotation.